pypdf/_reader.py

```diff
@@ -757,9 +757,17 @@ def _get_named_destinations(
         # TABLE 3.33 Entries in a name tree node dictionary (PDF 1.7 specs)
         elif CA.NAMES in tree:  # KIDS and NAMES are exclusives (PDF 1.7 specs p 162)
             names = cast(DictionaryObject, tree[CA.NAMES])
-            for i in range(0, len(names), 2):
+            i = 0
+            while i < len(names):
                 key = cast(str, names[i].get_object())
-                value = names[i + 1].get_object()
```

Comment:
Wouldn't it be simpler to  change the line `value = names[i + 1].get_object()` to

```
if len(names) < i + 1:
    break
value = names[i + 1].get_object()
```

Would that work?

Reply:
I wan to deal with cases where I can have [k1 v1 k2 v2 v3 k4 v4 k5 v5] so we need to check/load  items (key or value) one at a time and incrementing i at each step.
I agree that  we can change the code for L766-L769 to:
```
if  i>=len(names):
    break
values = names[i].get_object()
```
I choose my current writing [EAFP](https://docs.python.org/3.6/glossary.html#term-eafp) philosophy

Your recommendation ?

Reply:
Thanks for giving the example - I haven't thought about keys / values missing in between. Good job :+1: